audit-connector/src/main/java/com/sequenceiq/cloudbreak/audit/converter/AuditEventToGrpcAuditEventConverter.java

```diff
@@ -1,25 +1,49 @@
 package com.sequenceiq.cloudbreak.audit.converter;
 
-import static com.sequenceiq.cloudbreak.util.ConditionBasedEvaluatorUtil.doIfTrue;
-import static com.sequenceiq.cloudbreak.util.UuidUtil.uuidSupplier;
-
-import java.util.Optional;
-
-import org.apache.commons.lang3.StringUtils;
-import org.springframework.stereotype.Component;
-
 import com.cloudera.thunderhead.service.audit.AuditProto;
 import com.sequenceiq.cloudbreak.audit.model.ActorBase;
 import com.sequenceiq.cloudbreak.audit.model.ActorCrn;
 import com.sequenceiq.cloudbreak.audit.model.ActorService;
-import com.sequenceiq.cloudbreak.audit.model.ApiRequestData;
 import com.sequenceiq.cloudbreak.audit.model.AuditEvent;
 import com.sequenceiq.cloudbreak.audit.model.EventData;
-import com.sequenceiq.cloudbreak.audit.model.ServiceEventData;
+import com.sequenceiq.cloudbreak.util.NullUtil;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.sequenceiq.cloudbreak.util.ConditionBasedEvaluatorUtil.doIfTrue;
+import static com.sequenceiq.cloudbreak.util.UuidUtil.uuidSupplier;
 
 @Component
 public class AuditEventToGrpcAuditEventConverter {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuditEventToGrpcAuditEventConverter.class);
+
+    private final Map<Class, EventDataUtilizer> eventDataUtilizer;
+
+    public AuditEventToGrpcAuditEventConverter(Map<Class, EventDataUtilizer> eventDataUtilizer) {
+        if (MapUtils.isNotEmpty(eventDataUtilizer)) {
+            List<String> utilizers = new LinkedList<>();
+            eventDataUtilizer.forEach((type, utilizer) -> utilizers.add(String.format("[%s :: %s]",
+                    NullUtil.getIfNotNull(type, t -> t.getSimpleName()),
+                    NullUtil.getIfNotNull(utilizer, u -> u.getClass().getSimpleName()))));
+            String utilizerListMessage = String.join(",", utilizers);
```

Comment:
you could completely go functional and throw out the `utilizers` list. just map the k:v to output and use `Collectors.joining`

Reply:
I wouldn't go full functional just because it wouldn't add that much readability for me at least
![dont-go-full-functional](https://user-images.githubusercontent.com/16577123/84393128-7b2e6300-abfb-11ea-932d-59dc897dc448.jpg)

Reply:
well :D

Reply:
just do it, it will be beautiful. trust me ;)